drivers/flash/soc_flash_mcux.c

```diff
@@ -191,19 +191,26 @@ static int flash_mcux_read(const struct device *dev, off_t offset,
 	addr = offset + priv->pflash_block_base;
 
 #ifdef CONFIG_CHECK_BEFORE_READING
+  #ifdef CONFIG_SOC_LPC55S36
```

Comment:
Can we instead use a generic constant here that we `#define` based on the `compatible` string, or use the `compatible` string directly?

Reply:
Let's look at this FMC: as of now, it is only used on lpc55s36...